virttest/env_process.py

```diff
@@ -569,6 +569,64 @@ def preprocess(test, params, env):
     """
     error_context.context("preprocessing")
 
+    def _check_host_component_version(required_cpnt, cpnt, get_version_func):
+        """"
+        Check if a component on host satisfied the test component requirements.
+        If not, cancel the test.
+
+        :param required_cpnt: A tuple of required component.
+                              (None,"2.10.0") means the cpnt version must
+                              less then or equal to 2.10.0".
+                              ("2.1.0",None) means the cpnt version must
+                              greater then 2.1.0".
+                              ("2.1.0","2.10.0") means the cpnt version must
+                              less then or equal to 2.10.0" and greater then
+                              2.1.0".
+        :param cpnt: A raw component version string.
+        :param get_version_func: A function for getting actual version number.
+        """
+        from distutils.version import LooseVersion
+        host_component = get_version_func(cpnt)
+        required_cpnt = eval(required_cpnt)
+        func_info = get_version_func.__name__.replace("_", " ")
+        # successfully got the version string
+        if host_component:
```

Comment:
if not host_component:
    test.cancel()
    return
import operator as op
lookup = {0: op.le, 1: op.ge}
for i in range(2):
    if not required_cpnt[i]:
        continue
    flag = lookup[i](LooseVersion(required_cpnt[i]), LooseVersion(host_component))
    if not flag:
        test.cancel()
        return